elements/lisk-tree/test/sparse_merkle_tree/utils.spec.ts

```diff
@@ -283,4 +285,135 @@ describe('utils', () => {
 			expect(bufferToBinaryString(Buffer.from(data.buf, 'hex'))).toEqual(data.str);
 		});
 	});
+
+	describe('getOverlappingStr', () => {
+		it('returns valid overlapping strings', () => {
+			expect(getOverlappingStr('12356', '123456')).toEqual('123');
+			expect(getOverlappingStr('1010110', '1010010')).toEqual('1010');
+		});
+
+		it('returns empty string if no overlapping strings found', () => {
+			expect(getOverlappingStr('12356', '2356')).toEqual('');
+			expect(getOverlappingStr('0110011', '110011')).toEqual('');
+		});
+	});
+
+	describe('verify', () => {
+		it('should return false if number of query keys are not same as proof', () => {
+			const queryKeys = ['01111110', '01101101', '00011011'].map(binaryStringToBuffer);
+			const proof = {
+				siblingHashes: [
+					'e6fa536eaac055d524e29fb4682893e3111bf3a027f7cd5ba312aec56460eb1b',
+					'63a154f88e6f5898bada58cbcb0dfcfa84e18cd5d50783e6703d904bef8be36b',
+					'da7f3bd33f419f025fc34ada50e26bc0094a7f0018f91fa7e51b66c88c6a7e78',
+					'ed9b2d408363d6edec46b055de68e67b37091f5b7dece4415200082ba01bc73e',
+				].map(h => Buffer.from(h, 'hex')),
+				queries: [
+					createQueryObject({
+						key: 'e1',
+						value: 'f031efa58744e97a34555ca98621d4e8a52ceb5f20b891d5c44ccae0daaaa644',
+						bitmap: '17',
+					}),
+				],
+			};
+			const merkleRoot = Buffer.from(
+				'21ecda9db382eff32c9ec899fc7090cf58858e8c22a2af82510cd4d9c9a42c2f',
+				'hex',
+			);
+
+			expect(verify(queryKeys, proof, merkleRoot, 1)).toBeFalse();
+		});
+
+		it('should return false if queries does not match and proof does not provide inclusion proof of one of query key', () => {
```

Comment:
I think you mean to say here "proof does not provide inclusion or exclusion proof" since here the proof for the key `01100000` should be an exclusion proof.

Reply:
So the actual logic is either all `user queries keys` match to the `proof query objects keys`, or if they does not match then some of query object in the proof provide proof of the keys user asked. 

So in this test case we are checking the query objects asked by user `00110011, 01100000` are different than what we get in proof `00110011, 11100001`. In this case user asked for key `01100000` but proof contains `11100001` yet its good enough to provide inclusion of earlier key.